### PR TITLE
Add more type hints to AbstractAdmin

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is a pedantic change.

### CODE REVIEW

File: src/Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -3533,6 +3535,8 @@ protected function configureDefaultFilterValues(array &$filterValues)
      * Example:
      *   $sortValues['_sort_by'] = 'foo'
      *   $sortValues['_sort_order'] = 'DESC'
+     *
+     * @phpstan-param array{_sort_by?: string, _sort_order?: string} $sortValues
```
Comment (VincentLanglet) [inline src/Admin/AbstractAdmin.php:3539]:
```suggestion
     * @phpstan-param array{_page?: int, _per_page?: int, _sort_by?: string, _sort_order?: string} $sortValues
```

**Review by VincentLanglet — APPROVED** [review]

**Review by greg0ire — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

File: src/Admin/AbstractAdmin.php [hunk 2]
```diff
@@ -3533,6 +3535,8 @@ protected function configureDefaultFilterValues(array &$filterValues)
      * Example:
      *   $sortValues['_sort_by'] = 'foo'
      *   $sortValues['_sort_order'] = 'DESC'
+     *
+     * @phpstan-param array{_page?: int, _per_page?: int, _sort_by?: string, _sort_order?: string} $sortValues
```
Comment (jordisala1991) [inline src/Admin/AbstractAdmin.php:3539]: this could be simply param, afaik array shape is supported by Ides like phpstorm right @VincentLanglet

Comment (VincentLanglet) [reply src/Admin/AbstractAdmin.php:3539]: Phpstorm has a weird support about this.

image

It does understand array shape for return type.
But not fully for param type. (You can see that `$foo` is considered as a comment).

`@phpstan-param` seems to be more consistent with our new standard.

**Review by franmomu — APPROVED** [review]

File: src/Admin/AbstractAdmin.php (hunk 2 above)
Comment (core23) [reply src/Admin/AbstractAdmin.php:3539]: AFAIK this is no standard doc, so we should keep both annotations.

Comment (core23) [reply src/Admin/AbstractAdmin.php:3539]: The [RFC](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#arrays) does not mention this complex array information

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thank you @core23

**Pull request merged**